pkg/apis/networking/validation/validation.go

```diff
@@ -385,33 +396,64 @@ func validateHTTPIngressPath(path *networking.HTTPIngressPath, fldPath *field.Pa
 		}
 	}
 
-	allErrs = append(allErrs, validateIngressBackend(&path.Backend, fldPath.Child("backend"), opts)...)
+	allErrs = append(allErrs, validateIngressBackend(&path.Backend, fldPath.Child("backend"), opts, requestGV)...)
 	return allErrs
 }
 
+// portFieledName returns the name of the field used for an
+// ingress service port in the provided GroupVersion.
+func portFieldName(gv schema.GroupVersion) string {
+	switch gv {
+	case networkingv1beta1.SchemeGroupVersion, extensionsv1beta1.SchemeGroupVersion:
+		return "servicePort"
+	default:
+		return "port"
+	}
+}
+
+// serviceName returns the name of the field used for a
+// service in the provided GroupVersion.
+func serviceName(gv schema.GroupVersion) string {
+	switch gv {
+	case networkingv1beta1.SchemeGroupVersion, extensionsv1beta1.SchemeGroupVersion:
+		return "serviceName"
+	default:
+		return "name"
+	}
+}
+
 // validateIngressBackend tests if a given backend is valid.
-func validateIngressBackend(backend *networking.IngressBackend, fldPath *field.Path, opts IngressValidationOptions) field.ErrorList {
+func validateIngressBackend(backend *networking.IngressBackend, fldPath *field.Path, opts IngressValidationOptions, requestGV schema.GroupVersion) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	hasResourceBackend := backend.Resource != nil
-	hasServiceBackend := len(backend.ServiceName) > 0 || backend.ServicePort.IntVal != 0 || len(backend.ServicePort.StrVal) > 0
+	hasServiceBackend := backend.Service != nil
 
 	switch {
 	case hasResourceBackend && hasServiceBackend:
 		return append(allErrs, field.Invalid(fldPath, "", "cannot set both resource and service backends"))
 	case hasResourceBackend && !opts.allowResourceBackend:
 		return append(allErrs, field.Forbidden(fldPath.Child("resource"), "not supported; only service backends are supported in this version"))
-	case hasResourceBackend:
-		allErrs = append(allErrs, validateIngressTypedLocalObjectReference(backend.Resource, fldPath.Child("resource"))...)
-	default:
-		if len(backend.ServiceName) == 0 {
-			return append(allErrs, field.Required(fldPath.Child("serviceName"), ""))
+	case hasServiceBackend:
+		if backend.Service.Port.Name != "" && backend.Service.Port.Number != 0 {
+			return append(allErrs, field.Invalid(fldPath, "", "cannot set both port name & port number"))
 		}
-		for _, msg := range apivalidation.ValidateServiceName(backend.ServiceName, false) {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("serviceName"), backend.ServiceName, msg))
+		if backend.Service.Name == "" {
+			return append(allErrs, field.Required(fldPath.Child("name"), ""))
 		}
-		allErrs = append(allErrs, apivalidation.ValidatePortNumOrName(backend.ServicePort, fldPath.Child("servicePort"))...)
-
+		allErrs = append(allErrs, apivalidation.ValidateDNS1123Label(backend.Service.Name, fldPath.Child(serviceName(requestGV)))...)
```

Comment:
why the switch from apivalidation.ValidateServiceName to apivalidation.ValidateDNS1123Label?